### PR TITLE
rpasture patch

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

**Review by ryanchou-dev — APPROVED** [review]
this editorial is so hot

File: solutions/usaco-1063.mdx [hunk 1]
```diff
@@ -104,36 +158,36 @@ int main() {
...
+			StringTokenizer cow = new StringTokenizer(read.readLine());
+			cows[c][0] = Integer.parseInt(cow.nextToken());
+			cows[c][1] = Integer.parseInt(cow.nextToken());
+			assert !(seenX.contains(cows[c][0]) || seenY.contains(cows[c][1]));
 			seenX.add(cows[c][0]);
 			seenY.add(cows[c][1]);
 		}
 
 		// we do a little coordinate compression
 		Arrays.sort(cows, Comparator.comparingInt(c -> c[0]));  // sort by x
```
Comment (ChuangSheep) [inline solutions/usaco-1063.mdx:183]:
```suggestion
		// sort by x
		Arrays.sort(cows, Comparator.comparingInt(c -> c[0]));  
```

File: solutions/usaco-1063.mdx [hunk 2]
```diff
@@ -104,36 +158,36 @@ int main() {
...
-		HashMap<Integer, Integer> reducedX = new HashMap<>();
+		Map<Integer, Integer> reducedX = new HashMap<>();
 		for (int c = 0; c < cowNum; c++) {
 			reducedX.put(cows[c][0], c);
 		}
 
-		Arrays.sort(cows, Comparator.comparingInt(c -> c[1]));
-		HashMap<Integer, Integer> reducedY = new HashMap<>();  // sort by y
+		Arrays.sort(cows, Comparator.comparingInt(c -> c[1]));  // sort by y
```
Comment (ChuangSheep) [inline solutions/usaco-1063.mdx:189]:
```suggestion
 		// sort by y
		Arrays.sort(cows, Comparator.comparingInt(c -> c[1])); 
```

File: solutions/usaco-1063.mdx [hunk 3]
```diff
@@ -104,36 +158,36 @@ int main() {
...
 
 ```java
-import java.io.*;
 import java.util.*;
+import java.io.*;
 
-public final class RPasture {
-	public static void main(String[] args) throws IOException {
+public class RPasture {
+	public static void main (String[] args) throws IOException {
```
Comment (ChuangSheep) [inline solutions/usaco-1063.mdx:165]:
```suggestion
	public static void main(String[] args) throws IOException {
```
space?

**Review by ChuangSheep — APPROVED** [review]
well done (esp. the explanation)

File: solutions/usaco-1063.mdx [hunk 4]
```diff
@@ -30,19 +83,19 @@ using namespace std;
...
-		std::cin >> c.first >> c.second;
+		cin >> c.first >> c.second;
 		assert(!(seen_x.count(c.first) || seen_y.count(c.second)));
 		seen_x.insert(c.first);
 		seen_y.insert(c.second);
 	}
 
 	// we do a little coordinate compression
-	sort(cows.begin(), cows.end());  // sort by x first
+	sort(cows.begin(), cows.end());  // sort by x
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-1063.mdx:98]: as long as it doesn't exceed 80 chars it's fine

**Pull request merged**